security/security-design/fn-datasafe-dbaudit-to-oci-logging/terraform/modules/network/main.tf

```diff
@@ -0,0 +1,142 @@
+#OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#SOFTWARE.
+################################################################################
+
+resource "oci_core_virtual_network" "vcn" {
+  cidr_block     = var.VCN-CIDR
+  dns_label      = "${var.fnvcndnslabelprefix}${random_id.tag.hex}"
+  compartment_id = var.compartment_ocid
+  display_name   = "${var.fnvcnnameprefix}-${random_id.tag.hex}"
+}
+
+data "oci_core_services" "service_gateway_all_oci_services" {
+  filter {
+    name   = "name"
+    values = ["All .* Services In Oracle Services Network"]
+    regex  = true
+  }
+}
+
+resource "oci_core_service_gateway" "service_gw" {
+
+  compartment_id = var.compartment_ocid
+  vcn_id          = oci_core_virtual_network.vcn.id
+  services {
+        service_id = lookup(data.oci_core_services.service_gateway_all_oci_services.services[0], "id")
+  }
+  display_name = "${var.fnvcnnameservicegatewayprefix}-${random_id.tag.hex}"
+}
+
+resource "oci_core_route_table" "rt_fn_subnet" {
+  compartment_id = var.compartment_ocid
+  vcn_id         = oci_core_virtual_network.vcn.id
+  display_name   = "${var.fnvcnnameroutingtableprefix}-${random_id.tag.hex}"
+  
+  route_rules {
+    destination       = lookup(data.oci_core_services.service_gateway_all_oci_services.services[0], "cidr_block")
+    destination_type  = "SERVICE_CIDR_BLOCK"
+    network_entity_id = oci_core_service_gateway.service_gw.id
+    description = "${var.fnvcnroutingtabledescriptionervicegw}-${random_id.tag.hex}"
+  }
+  
+}
+
+
+
+resource "oci_core_dhcp_options" "dhcpoptions1" {
+  compartment_id = var.compartment_ocid
+  vcn_id         = oci_core_virtual_network.vcn.id
+  display_name   = "${var.fnvcnnamedhcpopitonsprefix}-${random_id.tag.hex}"
+  options {
+    type        = "DomainNameServer"
+    server_type = "VcnLocalPlusInternet"
+  }  
+}
+
+resource "oci_core_subnet" "function_ds_log_vcn" {
+  cidr_block        = var.fnsubnet-CIDR
+  display_name      = "${var.fnsubnetnameprefix}-${random_id.tag.hex}"
+  dns_label         = "${var.fnsubnetdnslabelprefix}${random_id.tag.hex}"
+  compartment_id    = var.compartment_ocid
+  vcn_id            = oci_core_virtual_network.vcn.id
+  route_table_id    = oci_core_route_table.rt_fn_subnet.id
+  dhcp_options_id   = oci_core_dhcp_options.dhcpoptions1.id
+  prohibit_internet_ingress = true
+  prohibit_public_ip_on_vnic = true
+  security_list_ids = [oci_core_security_list.vcn_security_list.id]
+}
+
+resource "oci_core_security_list" "vcn_security_list"{
+  compartment_id = var.compartment_ocid
+  vcn_id         = oci_core_virtual_network.vcn.id
+  display_name = "${var.fnvcnnamesecuritylistprefix}-${random_id.tag.hex}"
+
+  egress_security_rules {
+      stateless = false
+      destination = "0.0.0.0/0"
+      destination_type = "CIDR_BLOCK"
+      protocol = "all" 
+  }
+
+  ingress_security_rules { 
+      stateless = false
+      source = "0.0.0.0/0"
```

Comment:
Should be customizable.

Reply:
Resolved, I prefer to offer this default configuration.